cristin-organization/src/test/java/no/unit/nva/cristin/organization/FetchCristinOrganizationTest.java

```diff
@@ -0,0 +1,192 @@
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR;
+import static org.apache.http.HttpStatus.SC_NOT_FOUND;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasKey;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+class FetchCristinOrganizationTest {
+
+    public static final ObjectMapper restApiMapper = JsonUtils.dtoObjectMapper;
+    public static final String IDENTIFIER = "identifier";
+    public static final String IDENTIFIER_VALUE = "1.0.0.0";
+    public static final String ORGANIZATION_NOT_FOUND_MESSAGE = "Organization not found: ";
+    private static final String IDENTIFIER_NULL_ERROR = "Identifier is not a valid Organization identifier: null";
+    FetchCristinOrganizationHandler fetchCristinOrganizationHandler;
+    private CristinApiClient cristinApiClient;
+    private Environment environment;
+    private ByteArrayOutputStream output;
+    private Context context;
+
+    @BeforeEach
+    public void setUp() {
+
+        environment = mock(Environment.class);
+        when(environment.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn("*");
+        context = mock(Context.class);
+        cristinApiClient = new CristinApiClient();
+        output = new ByteArrayOutputStream();
+        fetchCristinOrganizationHandler = new FetchCristinOrganizationHandler(cristinApiClient, environment);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "1.0.0.0, true",
+            "1.0.0, false",
+            "1.0, false",
+            "1, false",
+            "John, false"
+    })    void identifierPatternMatcher(String identifier, boolean expected) {
+        final Pattern pattern = Pattern.compile(IDENTIFIER_PATTERN);
+        final boolean matches = pattern.matcher(identifier).matches();
+        final boolean valid  = matches == expected;
+        assertThat(identifier + " is " + valid, valid);
+    }
+
+    @Test
+    public void handlerReturnsNotFoundResponseOnPublicationMissing()
+            throws IOException, ApiGatewayException, InterruptedException {
+
+        cristinApiClient = spy(cristinApiClient);
+        doThrow(new NonExistingUnitError("Organization not found: " + IDENTIFIER_VALUE))
+                .when(cristinApiClient).getSingleUnit(any(), any());
+
+        fetchCristinOrganizationHandler = new FetchCristinOrganizationHandler(cristinApiClient, environment);
+        fetchCristinOrganizationHandler.handleRequest(generateHandlerRequest(IDENTIFIER_VALUE), output, context);
+        GatewayResponse<Problem> gatewayResponse = parseFailureResponse();
+
+        assertEquals(SC_NOT_FOUND, gatewayResponse.getStatusCode());
```

Comment:
I find I can generalise the basic testing to a couple of methods since they repeat a pattern